## 自定义错误处理

```js
func download(url string) (resp.Body,error)  {
    body, err := ioutil.ReadAll(resp.Body)
    if err != nil {
        fmt.Println("read response err",err)
        return nil, err
    }
      return resp.Body, nil 
}


func ParseErr(err error) {
    if err != nil {
        log(err)
        return
    }
}



// 如何能简化到这面的程度
func download(url string) (resp.Body,error)  {

    body, err := ioutil.ReadAll(resp.Body)
    ParseError(err)
    return resp.Body, nil 

}


```

